test/test_magnet.cpp

```diff
@@ -224,6 +224,23 @@ TORRENT_TEST(magnet)
 	p2 = s->abort();
 }
 
+TORRENT_TEST(parse_escaped_hash_parameter)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn%3Abtih%3Acdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd", p, ec);
```

Comment:
although correct, is there an actual use case for this? as far as I can tell, the intention behind the magnet link syntax is to not need to escape the colons.